lib/gitsh/prompter.rb

```diff
@@ -31,6 +31,10 @@ def prompt
 
     attr_reader :env, :prompt_color
 
+    def cwd_basename
+      Dir.getwd == Dir.home ? '~' : File.basename(Dir.getwd)
```

Comment:
Do you mean to do the replacement in the `%d` format string too? i.e. something like:
```diff
diff --git a/lib/gitsh/prompter.rb b/lib/gitsh/prompter.rb
index b650cb3..a977700 100644
--- a/lib/gitsh/prompter.rb
+++ b/lib/gitsh/prompter.rb
@@ -20,8 +20,8 @@ module Gitsh
         '%b' => branch_name,
         "%B" => shortened_branch_name,
         '%c' => status_color,
-        '%d' => Dir.getwd,
-        '%D' => File.basename(Dir.getwd),
+        '%d' => working_directory,
+        '%D' => File.basename(working_directory),
         '%w' => clear_color,
         '%#' => terminator
       })
@@ -31,6 +31,10 @@ module Gitsh
 
     attr_reader :env, :prompt_color
 
+    def working_directory
+      Dir.getwd.sub(/\A#{Dir.home}/, '~')
+    end
+
     def padded_prompt_format
       "#{prompt_format.chomp} "
     end
```